specification/web/resource-manager/Microsoft.Web/stable/2018-02-01/WebApps.json

```diff
@@ -18918,6 +20271,65 @@
         }
       }
     },
+    "FunctionStatus": {
+      "description": "Function status.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "FunctionStatus resource specific properties",
+          "properties": {
+            "errors": {
+              "description": "Collection of initialization errors for the function.",
+              "type": "array",
+              "items": {
+                "type": "string"
+              }
+            }
+          },
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "HostKeys": {
+      "description": "Functions host level keys.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "./CommonDefinitions.json#/definitions/ProxyOnlyResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "HostKeys resource specific properties",
+          "properties": {
+            "masterKey": {
```

Comment:
This structure suggests the response from the API will have a body with the structure:

```json
{
 "properties": {
     "masterKey": "thingy"
 }
}
```

Looking at a request to this API made by the functions blade in the portal the response isn't nested in a `properties` object.

Request Headers:
```
POST /subscriptions/5774ad8f-d51e-4456-a72e-0447910568d3/resourceGroups/lgt1/providers/Microsoft.Web/sites/lgt1/host/default/listkeys?api-version=2015-08-01 HTTP/1.1
Host: management.azure.com
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0
Accept: application/json
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
Content-Type: application/json
Authorization: Bearer REDACTED
x-ms-client-request-id: a2894cd2-a77e-45c1-9648-a9dda8321563
x-ms-client-session-id: 58da440e458f468d885fb9758aaa987b
Content-Length: 0
Origin: https://functions.azure.com
DNT: 1
Connection: keep-alive
Referer: https://functions.azure.com/ng-min/resources/subscriptions/5774ad8f-d51e-4456-a72e-0447910568d3/resourcegroups/lgt1/sites/lgt1?correlationId=83b742d8-bb51&sessionId=58da440e458f468d885fb9758aaa987b&frameId=45517dae-8204&ElasticScaleOut=true&appsvc.bladetype=appblade&trustedAuthority=https:%2F%2Fms.portal.azure.com
Pragma: no-cache
Cache-Control: no-cache
```
Response Body:
```
{
 "masterKey":"Gfbg2Tv4lwZW/a8fLZzLr7sgQYmHau9rQWERl4yG4P34REDACTED",
 "functionKeys":{"default":"Sr4Kth4Io86N1OOTEXhkJeqq99bN2TWLXREDACTED=="},
 "systemKeys":{}
}
```

Reply:
Do you see the same result if you try one of the existing APIs not modified in this PR, e.g. api/sites/{name}[/slots/{slot}]/functions/{functionName}/listsecrets? That API doesn't return a nested result either - it returns direct results. You'll note that both that return type (FunctionSecrets) and the type for this new API (HostKeys) follow the same type definition pattern.

Reply:
> Do you see the same result if you try one of the existing APIs not modified in this PR

I have a PR open here to resolve the same issue in an already existing call `WebApps_ListFunctionSecrets` https://github.com/Azure/azure-rest-api-specs/pull/7155. 

> In this PR I'm consistently following the pattern of our other existing APIs.

This likely indicates a wider issue with the way in which you are generating the specs or that I have made a mistake in my testing. Where you able to validate my sample request shared above is accurate?

Reply:
fixed